### DISCUSSION TITLE
Use an async database url

### DISCUSSION THREAD

**Discussion by italodeandra** [nextauthjs/next-auth#1810] [Help] [answered]
I need to do something like this:

```ts
export default NextAuth({
  providers: [
    Providers.GitHub({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET
    }),
  ],
  database: await getDatabaseConnection(),
})
```

Is it by any means, possible with NextAuth?

**Comment by balazsorban44** [comment, marked as answer]
Yes, totally!
```js
export default async function nextAuthHandler(req, res) {
  return NextAuth(req, res, {
    providers: [
      Providers.GitHub({
        clientId: process.env.GITHUB_ID,
        clientSecret: process.env.GITHUB_SECRET
      }),
    ],
    database: await getDatabaseConnection(),
  })
}
```

At the end of the day, `NextAuth` is just a nice wrapper for a Next.js API route handler :slightly_smiling_face:

The currently advertised way was introduced in https://github.com/nextauthjs/next-auth/issues/868

but you probably have seen many tutorials/people still doing:
```js
export default (req, res) => NextAuth(req, res, options)
```

The new way is just easier on the eye in my opinion, so I try to use it whenever possible.

**Reply by italodeandra** [reply]
I tried that, but `typescript` was a bad boy and didn't let me.

Now I just look dumb because I tried again and it worked.

Thanks man ❤️

**Reply by balazsorban44** [reply]
As a JS developer, at the end of the day, don't let TS be the boss! 🤷‍♂️😎

**Reply by italodeandra** [reply]
I started recently in TypeScript (2 years only). A while ago I tried to not use anymore.

It didn't take long before I started missing the code completion and trying to use JSDoc, and then getting pissed on how difficult it is to write and how bad the code looks afterwards. Then I went back to TypeScript.

You seem to have more experience with not letting the TS be the boss. Can you help me? 😂
